### PR TITLE
feat: create docker images for client

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

The new workflow shows how this can be used, but to surface the important parts:

Build:

```
docker build \
            --build-arg HOME_LOCATION=http://localhost:8000 \
            --build-arg API_LOCATION=http://localhost:3000 \
            --build-arg FORUM_LOCATION=https://forum.freecodecamp.org \
            --build-arg NEWS_LOCATION=https://www.freecodecamp.org/news \
            --build-arg RADIO_LOCATION=https://coderadio.freecodecamp.org \
            --build-arg CLIENT_LOCALE=english \
            --build-arg CURRICULUM_LOCALE=english \
            --build-arg SHOW_LOCALE_DROPDOWN_MENU=false \
            --build-arg ALGOLIA_APP_ID=app_id_from_algolia_dashboard \
            --build-arg ALGOLIA_API_KEY=api_key_from_algolia_dashboard \
            --build-arg STRIPE_PUBLIC_KEY=pk_from_stripe_dashboard \
            --build-arg PAYPAL_CLIENT_ID=id_from_paypal_dashboard \
            --build-arg PATREON_CLIENT_ID=id_from_patreon_dashboard \
            --build-arg DEPLOYMENT_ENV=staging \
            --build-arg SHOW_UPCOMING_CHANGES=false \
            --build-arg SHOW_NEW_CURRICULUM=true \
            --build-arg GROWTHBOOK_URI=api_URI_from_Growthbook_dashboard \
            --build-arg FREECODECAMP_NODE_ENV=development \
            -t fcc-client \
            -f docker/web/Dockerfile .
```

Start:

```
docker compose up
```

To control the host port:

```
HOST_PORT=12345 docker compose up
```

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
@raisedadead you can see the thought process here and I've left comments about what I think should be done.

Feel free to push any commits you see fit. Otherwise I'll get on with it when I can.

**Review by raisedadead — CHANGES_REQUESTED** [review]

**Comment by raisedadead** [comment]
Are the shell scripts meant to be placed along side the container, and mounted as volumes?

**Comment by ojeytonwilliams** [comment]
The first time you `docker compose up` is a little noisy:

<details>
<summary> logs </summary>

```
✔ Container freecodecamp-client-1  Recreated                                                                                                                                                                 0.1s 
Attaching to freecodecamp-client-1
freecodecamp-client-1  | 2023-08-31T13:18:16: PM2 log: Launching in no daemon mode
freecodecamp-client-1  | 2023-08-31T13:18:16: PM2 log: App [client-primary:0] starting in -fork mode-
freecodecamp-client-1  | 2023-08-31T13:18:16: PM2 log: App [client-primary:0] online
freecodecamp-client-1  | 2023-08-31T13:18:16: PM2 log: App [client-primary:1] starting in -fork mode-
freecodecamp-client-1  | 2023-08-31T13:18:16: PM2 log: App [client-primary:1] online
freecodecamp-client-1  | 2023-08-31T13:18:16: PM2 log: App [client-primary:2] starting in -fork mode-
freecodecamp-client-1  | 2023-08-31T13:18:16: PM2 log: App [client-primary:2] online
freecodecamp-client-1  | 2023-08-31T13:18:16: PM2 log: App [client-primary:3] starting in -fork mode-
freecodecamp-client-1  | 2023-08-31T13:18:16: PM2 log: App [client-primary:3] online
freecodecamp-client-1  | node:internal/process/promises:288
freecodecamp-client-1  |             triggerUncaughtException(err, true /* fromPromise */);
freecodecamp-client-1  |             ^
freecodecamp-client-1  | [Error: EEXIST: file already exists, mkdir '/tmp/update-check'] {
freecodecamp-client-1  |   errno: -17,
freecodecamp-client-1  |   code: 'EEXIST',
freecodecamp-client-1  |   syscall: 'mkdir',
freecodecamp-client-1  |   path: '/tmp/update-check'
freecodecamp-client-1  | }
freecodecamp-client-1  | Node.js v18.17.1
freecodecamp-client-1  | node:internal/process/promises:288
freecodecamp-client-1  |             triggerUncaughtException(err, true /* fromPromise */);
freecodecamp-client-1  |             ^
freecodecamp-client-1  | [Error: EEXIST: file already exists, mkdir '/tmp/update-check'] {
freecodecamp-client-1  |   errno: -17,
freecodecamp-client-1  |   code: 'EEXIST',
freecodecamp-client-1  |   syscall: 'mkdir',
freecodecamp-client-1  |   path: '/tmp/update-check'
freecodecamp-client-1  | }
freecodecamp-client-1  | Node.js v18.17.1
freecodecamp-client-1  | node:internal/process/promises:288
freecodecamp-client-1  |             triggerUncaughtException(err, true /* fromPromise */);
freecodecamp-client-1  |             ^
freecodecamp-client-1  | [Error: EEXIST: file already exists, mkdir '/tmp/update-check'] {
freecodecamp-client-1  |   errno: -17,
freecodecamp-client-1  |   code: 'EEXIST',
freecodecamp-client-1  |   syscall: 'mkdir',
freecodecamp-client-1  |   path: '/tmp/update-check'
freecodecamp-client-1  | }
freecodecamp-client-1  | Node.js v18.17.1
freecodecamp-client-1  | 2023-08-31T13:18:16: PM2 log: App [client-primary:2] exited with code [1] via signal [SIGINT]
freecodecamp-client-1  | 2023-08-31T13:18:16: PM2 log: App [client-primary:1] exited with code [1] via signal [SIGINT]
freecodecamp-client-1  | 2023-08-31T13:18:16: PM2 log: App [client-primary:3] exited with code [1] via signal [SIGINT]
freecodecamp-client-1  | 2023-08-31T13:18:16: PM2 log: App [client-primary:2] starting in -fork mode-
freecodecamp-client-1  | 2023-08-31T13:18:16: PM2 log: App [client-primary:2] online
freecodecamp-client-1  | 2023-08-31T13:18:16: PM2 log: App [client-primary:1] starting in -fork mode-
freecodecamp-client-1  | 2023-08-31T13:18:16: PM2 log: App [client-primary:1] online
freecodecamp-client-1  | 2023-08-31T13:18:16: PM2 log: App [client-primary:3] starting in -fork mode-
freecodecamp-client-1  | 2023-08-31T13:18:16: PM2 log: App [client-primary:3] online
freecodecamp-client-1  | UPDATE AVAILABLE The latest version of `serve` is 14.2.1
freecodecamp-client-1  | INFO: Discovered configuration in `../serve.json`
freecodecamp-client-1  | INFO: Accepting connections at http://localhost:8000
freecodecamp-client-1  | UPDATE AVAILABLE The latest version of `serve` is 14.2.1
freecodecamp-client-1  | INFO: Discovered configuration in `../serve.json`
freecodecamp-client-1  | UPDATE AVAILABLE The latest version of `serve` is 14.2.1
freecodecamp-client-1  | INFO: Discovered configuration in `../serve.json`
freecodecamp-client-1  | UPDATE AVAILABLE The latest version of `serve` is 14.2.1
```

</details>

It's mostly `serve` looking to see if there's a new version, so I don't think there's a real problem (just noise).

File: docker/web/Dockerfile [hunk 1]
```diff
@@ -1,28 +1,58 @@
...
+RUN pnpm install -w -F=client -F=ui -F=browser-scripts -F=challenge-parser \
+--frozen-lockfile
+RUN pnpm build:client
+
+FROM node:18-bullseye AS client-config
 USER node
 WORKDIR /home/node
-COPY --from=builder /home/node/build/client/public/ client/public
-COPY --from=builder /home/node/build/docker/web/serve.json client/serve.json
+RUN git clone --depth=1 https://github.com/freeCodeCamp/client-config.git client
```
[the remark this answers is not in the document]
Comment (ojeytonwilliams) [reply docker/web/Dockerfile]: Makes sense. It was already there, so I just updated it to the current version in client-config.

**Comment by raisedadead** [comment]
Remind me: We can't use the next version of serve because?

**Review by raisedadead — APPROVED** [review]

**Pull request merged**